### DISCUSSION TITLE
Debugging custom tokenizer?

### DISCUSSION THREAD

**Discussion by erotavlas** [explosion/spaCy#8890] [Help: Coding & Implementations] [answered]
Do you have any ideas how I can debug a custom tokenizer script?  In spacy 2.x I used to create an empty  base model with the modified tokenizer rules and load that and test my strings with it.

how can I do something similar with spacy 3.x but without having to train a model to embed the tokenizer into it?

```
import pytest

def tokenize(inputstring, spacy_model):
    tokens = []
    doc = model(inputstring)
    for t in doc:
        tokens.append(t)
    
    return tokens

def testcase_8():
    
    tokens = tokenize('Inc.', nlp)
    assert len(tokens) == 2
    assert tokens[0].text == 'Inc'
    assert tokens[1].text == '.'


```

**Comment by adrianeboyd** [comment, marked as answer]
Create a config with just the tokenizer (to clarify: a normal full config, but with no components) and use `spacy assemble` to create a model from it. Then you can evaluate it however you evaluate your base models (e.g.,  with `spacy evaluate`).
